File: src/main/java/ru/romangr/catbot/executor/action/TelegramAction.java
package ru.romangr.catbot.executor.action;

import ru.romangr.catbot.telegram.model.Chat;
import ru.romangr.catbot.telegram.model.ExecutionResult;
import ru.romangr.exceptional.Exceptional;

public interface TelegramAction {

  Exceptional<ExecutionResult> execute();

  Chat getChat();

}

File: src/main/java/ru/romangr/catbot/executor/action/TelegramAction.kt
package ru.romangr.catbot.executor.action

import ru.romangr.catbot.telegram.model.Chat
import ru.romangr.catbot.telegram.model.ExecutionResult
import ru.romangr.exceptional.Exceptional

interface TelegramAction {

    val chat: Chat

    fun execute(): Exceptional<ExecutionResult>

}
